### PR TITLE
[IMP] project_timesheet_holidays: add redirect warning

### PR DESCRIPTION
**Pull request by kcv-odoo**
This PR add redirect warning with view time off button when
user try to delete timesheets that are linked to time off which
redirect to timesheet related time off and make timesheet page
readonly for tasks that are linked to leave type

task-2888607

### CODE REVIEW

File: addons/project_timesheet_holidays/models/account_analytic.py [hunk 1]
```diff
@@ -11,7 +11,28 @@ class AccountAnalyticLine(models.Model):
     holiday_id = fields.Many2one("hr.leave", string='Leave Request')
     global_leave_id = fields.Many2one("resource.calendar.leaves", string="Global Time Off", ondelete='cascade')
 
+    def _get_redirect_action(self):
+        action_data = {
+           'name': _('Time Off'),
+           'type': 'ir.actions.act_window',
+           'res_model': 'hr.leave',
+        }
+        if len(self.mapped("holiday_id")) == 1:
```
Comment (auon-odoo) [inline addons/project_timesheet_holidays/models/account_analytic.py:20]:
```suggestion
        if len(self.holiday_id) == 1:
```

File: addons/project_timesheet_holidays/models/account_analytic.py [hunk 2]
```diff
@@ -11,7 +11,28 @@ class AccountAnalyticLine(models.Model):
...
+            action_data.update({
+                'res_id': self.mapped("holiday_id")[0].id,
```
Comment (auon-odoo) [inline addons/project_timesheet_holidays/models/account_analytic.py:22]:
```suggestion
                'res_id': self.holiday_id.id,
```

File: addons/project_timesheet_holidays/models/account_analytic.py [hunk 3]
```diff
@@ -11,7 +11,28 @@ class AccountAnalyticLine(models.Model):
...
+                'views': [(self.env.ref('hr_holidays.hr_leave_view_form').id, 'form')],
```
Comment (auon-odoo) [inline addons/project_timesheet_holidays/models/account_analytic.py:23]: Perhaps `self.env.ref('hr_holidays.hr_leave_view_form').id` is worth a variable?

**Review by auon-odoo — COMMENTED** [review]
Hello :wave: 
Nice work! I left some comments.

File: addons/project_timesheet_holidays/models/account_analytic.py (hunk 1 above)
Comment (kcv-odoo) [reply addons/project_timesheet_holidays/models/account_analytic.py]: Done!! All the changes. Thanks :)

Comment (auon-odoo) [reply addons/project_timesheet_holidays/models/account_analytic.py]: Cool, thank you :+1:

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂 
Just a suggestion. Let me know whether you change or not.

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂 
@robodoo r+

**Comment by IT-Ideas** [comment]
@robodoo rebase-ff

**Pull request closed without merging**